docs/book/user-guide/tutorial/trigger-pipelines-from-external-systems.md

```diff
@@ -0,0 +1,881 @@
+    return {
+        "X_train": X_train,
+        "X_test": X_test,
+        "y_train": y_train,
+        "y_test": y_test,
+    }
+
+@step
+def train_model(
+    datasets: Dict[str, Any], model_type: str = "random_forest"
+) -> Union[RandomForestClassifier, GradientBoostingClassifier]:
+    """Train a model based on the specified type."""
+    X_train = datasets["X_train"]
+    y_train = datasets["y_train"]
+    
+    if model_type == "random_forest":
+        model = RandomForestClassifier(n_estimators=100, random_state=42)
+    elif model_type == "gradient_boosting":
+        model = GradientBoostingClassifier(random_state=42)
+    else:
+        raise ValueError(f"Unknown model type: {model_type}")
+    
+    print(f"Training a {model_type} model...")
+    model.fit(X_train, y_train)
+    return model
+
+@step
+def evaluate(
+    datasets: Dict[str, Any],
+    model: Union[RandomForestClassifier, GradientBoostingClassifier],
+) -> Dict[str, float]:
+    """Evaluate the model and return metrics."""
+    X_test = datasets["X_test"]
+    y_test = datasets["y_test"]
+    
+    y_pred = model.predict(X_test)
+    accuracy = accuracy_score(y_test, y_pred)
+    
+    print(f"Model accuracy: {accuracy:.4f}")
+    return {"accuracy": float(accuracy)}
+
+# Define Docker settings for the pipeline
+docker_settings = DockerSettings(
+    requirements="requirements.txt",
+    required_integrations=["sklearn"],
+    python_package_installer="uv",
+)
+
+@pipeline(settings={"docker": docker_settings})
+def training_pipeline(
+    data_url: str = "example-data-source", model_type: str = "random_forest"
+):
+    """A configurable training pipeline that can be triggered externally."""
+    data = load_data(data_url)
+    datasets = preprocess(data)
+    model = train_model(datasets, model_type)
+    metrics = evaluate(datasets, model)
+    return metrics
+```
+
+### Creating a Requirements File
+
+Create a `requirements.txt` file with the necessary dependencies:
+
+```plaintext
+# Requirements for pipeline trigger API
+fastapi>=0.95.0
+uvicorn>=0.21.0
+requests>=2.28.0
+# Core dependencies
+scikit-learn>=1.0.0
+pandas>=1.3.0
+numpy>=1.20.0
+# ZenML
+zenml>=0.80.1
+```
+
+### Creating a FastAPI Wrapper
+
+Next, create the `pipeline_api.py` file with the FastAPI application:
+
+```python
+import os
+import sys
+import importlib.util
+from typing import Dict, Any, Optional
+import threading
+from fastapi import FastAPI, HTTPException, Depends, Security
+from fastapi.security import APIKeyHeader
+from pydantic import BaseModel
+import uvicorn
+
+# Import the training pipeline from the common module
+from common import training_pipeline
+
+# Setup FastAPI app
+app = FastAPI(title="ZenML Pipeline Trigger API")
+
+# Simple API key authentication
+# This environment variable serves as a security token to protect your API endpoints
+# In production, use a strong, randomly generated key stored securely
+API_KEY = os.environ.get("PIPELINE_API_KEY", "your-secure-api-key")
```

Comment:
Do we really need this? seems unneccssary

Reply:
I think it's good to demonstrate it yes.